### ISSUE TITLE
Saw you would like me to help on building a new integration for Home Assistant

### ISSUE THREAD

**Issue by altersis** [dkingfx/Autonomic-MMS#4]
Hi there, just saw your invitation to work with you on this new integration, and would be happy to help. Could you please let me know what kind of help you're looking for? I was the last person to fix the Russound RNET integration, and understand the HASS development process well. Are you an Autonomic customer or are you related somehow with the company? 

cheers!

-Altersis

**Comment by dkingfx** [comment]
Hi Altersis,

I'm an Autonomic Mirage customer and own the eSeries 4 zones amp and two sources media streamer. I'm trying to start with a basic API similar to Russound RNET and at the very least have the component show up in HASS. I understand this may be a more complex project/task.

There are so many features included with the autonomic devices that it would take a very long time to code them all. I tried manipulating the Russound RNET Github files and Home Assistant modules to get the basic connection working to no avail.

could you help me develop a basic API and a test_harness that would send a ping/play command?

Unless you think that using the Russound RNET project to get Autonomic to work is not the right to move? The tcp_echo_client.py and tcp_socket_client.py both send the play command and music starts playing, not sure which method would be best.

Also, any guides on developing Home Assistant components would be greatly appreciated.

Thanks for taking the time to respond to the request.

-Dan

**Comment by altersis** [comment]
Got it. I took a quick peek at the Autonomic website, and yup, their devices look interesting. From what you are describing, In my opinion, the minimum viable product would have the following features:

**Per Zone:**
1. Turn on/off 
2. Control volume and mute
3. Allow selecting a source 

**Per Source:**
1. Turn on/off
2. Select what to play by selecting a favorite or a playlist

Is this what you would like to achieve at the beginning? I saw a bunch of documentation uploaded into GitHub, so I want to believe that all is relevant to your system, so I'll take a look at it during the weekend, and once I get your confirmation from the above, we'll plan next steps.
Cheers!

**Comment by dkingfx** [comment]
Hi,

Thanks for getting back to me. I've been playing with many files and configs trying to get something going. At the moment, I was able to send a basic play command using the test_harness.py and basic_api.py.

I think your suggestion would be the way to go, as there are many commands that the system respond to on a per-port basis (5004, 4006, 23), we really won't' need the features of the port 23 (reboot, shutdown, etc.)

Digging around Home Assistant media_players files, I came across the Pioneer file that sends telnet command to a host. At the moment, I'm getting playback, pause, volume up and down and next track and previous track. I will be uploading this file tot he test folder.

I would like to achieve a product that's able to control features from both telnet ports (5004, 5004) so yes, what you're suggesting would be a great start to this project.

Thanks again, I really appreciate your support!!!!

Thanks
Dan
